docs/guides/nodejs/survey-application.mdx

```diff
@@ -0,0 +1,365 @@
+
+    receipt.end()
+  })
+
+  return await doneWriting
+}
+```
+
+This PDF output is fairly plain, you can enhance it further using layout templates or branding.
+
+## Step 4: Delivery Logic
+
+Simulate or perform delivery of the receipt (e.g. via email or other downstream systems).
+
+```ts title:services/deliver.ts
+import { receipts } from '../resources/resources'
+
+receipts.subscribe(async (ctx) => {
+  const { id } = ctx.req.json()
+
+  // Simulate delivery or hook into a real email/SaaS integration
+  console.log(`Delivering receipt for submission: ${id}`)
+})
+```
+
+## Step 5: Retrieve Receipts
+
+Create an endpoint that returns a download URL for the generated receipt file.
+
+```ts title:services/receipts.ts
+import { api } from '@nitric/sdk'
+import { output } from '../resources/resources'
+
+const receiptApi = api('receipts')
+
+receiptApi.get('/receipts/:id', async (ctx) => {
+  const id = ctx.req.params.id
+  const file = output.file(`${id}.pdf`)
+  const url = await file.getDownloadUrl()
+  ctx.res.body = url
+})
+```
+
+## Run and Test Locally
+
+```bash
+nitric start
+```
+
+### Submit a Survey
+
+Use the dashboard to submit your survey data -
+
+Form ID: feedback
+
+```json
+{
+  "name": "Jane",
+  "rating": 5,
+  "feedback": "Great experience!"
+}
+```
+
+Form ID: rsvp
+
+```json
+{
+  "name": "Jane",
+  "attending": true,
+  "guests": 5
+}
+```
+
+![Submit Form](/docs/images/guides/survey-application/submit.png)
+
+You should see messages in your console for PDF generation and delivery.
+
+```bash
+Delivering receipt for submission: rsvp-1744131025791
+Receipt stored for rsvp-1744131025791
+```
+
+### Get a Receipt
+
+Replace `<timestamp>` with the value returned from the submission.
+
+![Get receipt](/docs/images/guides/survey-application/receipt.png)
+
+Use the URL in the response to retrieve your PDF:
+
+![Get PDF](/docs/images/guides/survey-application/pdf.png)
+
+## Deploying to AWS
+
+### Create your stack
+
+Create an AWS stack called `aws-staging` for your staging environment.
+
+```bash
+nitric stack new aws-staging aws
+```
+
+Inside the stack file, ensure you set your `region`.
+
+```yaml title:nitric.dev.yaml
+provider: nitric/aws@latest
+region: us-east-2
+```
+
+### Deploy
+
+Deploy to AWS using the `nitric up` command. Ensure you have set up your [AWS credentials](/providers/pulumi/aws#usage) correctly.
+
+```bash
+nitric up
+```
+
+### Tear down
+
+To avoid unwanted costs of running your test app, you can tear down the stack using the `nitric down` command.
+
+```bash
+nitric down
+```
+
+## Summary
+
+In this guide, you built a backend for handling survey submissions using Nitric and TypeScript:
+
+- Created a REST API to accept form submissions
+- Validated user input using Zod
+- Stored submissions in a key-value store
+- Generated PDF receipts using pdfkit
+- Stored the receipts in cloud storage
+- Delivered them asynchronously via event topics
+- Exposed a URL endpoint to retrieve generated receipts
```

Comment:
link to the website guide in a whats next heading possibly?

Reply:
done.